lib/numorstring/port.go

```diff
@@ -18,52 +18,79 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"regexp"
 	"strconv"
-	"strings"
 )
 
+// Port represents either a range of numeric ports or a named port.
+//
+//     - For a named port, set the PortName, leaving MinPort and MaxPort as 0.
+//     - For a port range, set MinPort and MaxPort to the (inclusive) port numbers.  Set
+//       PortName to "".
+//     - For a single port, set MinPort = MaxPort and PortName = "".
 type Port struct {
-	MinPort uint16
-	MaxPort uint16
+	MinPort  uint16
+	MaxPort  uint16
+	PortName string
 }
 
 // SinglePort creates a Port struct representing a single port.
 func SinglePort(port uint16) Port {
-	return Port{port, port}
+	return Port{MinPort: port, MaxPort: port}
+}
+
+func NamedPort(name string) Port {
+	return Port{PortName: name}
 }
 
 // PortFromRange creates a Port struct representing a range of ports.
 func PortFromRange(minPort, maxPort uint16) (Port, error) {
-	port := Port{minPort, maxPort}
+	port := Port{MinPort: minPort, MaxPort: maxPort}
 	if minPort > maxPort {
 		msg := fmt.Sprintf("minimum port number (%d) is greater than maximum port number (%d) in port range", minPort, maxPort)
 		return port, errors.New(msg)
 	}
 	return port, nil
 }
 
+var (
+	allDigits = regexp.MustCompile(`^\d+$`)
+	portRange = regexp.MustCompile(`^(\d+):(\d+)$`)
+	nameRegex = regexp.MustCompile("^[a-zA-Z0-9_.-]{1,128}$")
```

Comment:
Is there no restriction such as the first character being alpha?  As this stands, I believe a number would also match nameRegex.

Reply:
I think that'd be sensible if we'd done it on day one but now it'd be a breaking change in the existing places we use names.